src/random_test.cpp

```diff
@@ -285,33 +285,65 @@ int sum_of_all_server_options() {
 
 /* pick some algorith */
 inline static std::string pick_algorithm_lock() {
-  /* pick algorith for current sql */
+  /* pick algorithm and lock for current sql */
   static auto lock = opt_string(LOCK);
   static auto algorithm = opt_string(ALGORITHM);
-  std::string locks[] = {"DEFAULT", "EXCLUSIVE", "SHARED", "NONE"};
-  std::string algorithms[] = {"INPLACE", "COPY", "INSTANT", "DEFAULT"};
+
+  std::vector<std::string> locks;
+  std::vector<std::string> algorithms;
+
   std::string current_lock;
   std::string current_algo;
-  if (lock.compare("all") == 0 && algorithm.compare("all") == 0) {
-    auto lock_index = rand_int(3);
-    auto algo_index = rand_int(3);
-    /* lock=none;algo=inplace|copy not supported */
-    if (lock_index == 3 && algo_index < 2)
-      lock_index = 0;
-    current_lock = locks[lock_index];
-    current_algo = algorithms[algo_index];
-  } else if (lock.compare("all") == 0) {
-    auto lock_index = rand_int(3);
-    current_lock = locks[lock_index];
-    current_algo = algorithm;
-  } else if (algorithm.compare("all") == 0) {
-    auto algo_index = rand_int(3);
-    current_algo = algorithms[algo_index];
-    current_lock = lock;
+
+  if (lock.compare("all") == 0) {
+    locks.push_back("DEFAULT");
+    locks.push_back("EXCLUSIVE");
+    locks.push_back("SHARED");
+    locks.push_back("NONE");
   } else {
-    current_lock = lock;
-    current_algo = algorithm;
+    std::transform(lock.begin(), lock.end(), lock.begin(),
```

Comment:
This calculation would be done every time we call pick_algorithm_lock. so move this outside of this block

Reply:
Fixed